File: articles/relativity-and-the-reconstruction-of-space-and-time/fortran/lorentz_factor_table.f90
! Lorentz Factor Table
!
! This Fortran workflow computes:
!
!   gamma = 1 / sqrt(1 - beta^2)
!
! and scaled kinetic energy:
!
!   K / (mc^2) = gamma - 1

program lorentz_factor_table
  implicit none

  integer :: i
  real(8), dimension(7) :: betas
  real(8) :: beta
  real(8) :: gamma
  real(8) :: relativistic_ke_scaled
  real(8) :: newtonian_ke_scaled

  betas = (/0.0d0, 0.1d0, 0.3d0, 0.5d0, 0.8d0, 0.9d0, 0.99d0/)

  print *, "beta,gamma,relativistic_ke_scaled,newtonian_ke_scaled"

  do i = 1, size(betas)
     beta = betas(i)
     gamma = 1.0d0 / sqrt(1.0d0 - beta**2)
     relativistic_ke_scaled = gamma - 1.0d0
     newtonian_ke_scaled = 0.5d0 * beta**2

     print '(F10.6,A,F14.8,A,F14.8,A,F14.8)', beta, ",", gamma, ",", relativistic_ke_scaled, ",", newtonian_ke_scaled
  end do

end program lorentz_factor_table

File: articles/relativity-and-the-reconstruction-of-space-and-time/c/lorentz_factor_table.c
/*
Lorentz Factor Table

This C workflow computes:

    gamma = 1 / sqrt(1 - beta^2)

and scaled kinetic energy:

    K / (mc^2) = gamma - 1
*/

#include <math.h>
#include <stdio.h>

double lorentz_factor(double beta) {
    return 1.0 / sqrt(1.0 - beta * beta);
}

int main(void) {
    double betas[] = {0.0, 0.1, 0.3, 0.5, 0.8, 0.9, 0.99};
    int n = sizeof(betas) / sizeof(betas[0]);

    printf("beta,gamma,relativistic_ke_scaled,newtonian_ke_scaled\n");

    for (int i = 0; i < n; i++) {
        double beta = betas[i];
        double gamma = lorentz_factor(beta);
        double relativistic_ke_scaled = gamma - 1.0;
        double newtonian_ke_scaled = 0.5 * beta * beta;

        printf("%.6f,%.8f,%.8f,%.8f\n",
               beta,
               gamma,
               relativistic_ke_scaled,
               newtonian_ke_scaled);
    }

    return 0;
}
